src/__tests__/init.test.ts

```diff
@@ -0,0 +1,103 @@
+import * as path from "path";
+import * as fse from "fs-extra";
+import { execSync } from "child_process";
+import { DEFAULT_COMPONENT_NAME } from "../commands/generate";
+
+let root = process.cwd();
+
+describe("The `init` command", () => {
+  const backupJson = "backup.json";
+  const backupYarnLock = "backupYarn.lock";
+
+  beforeAll(async () => {
+    try {
+      // Back up our package.json so we don't break anything during our tests
+      await fse.renameSync(`${root}/package.json`, `${root}/${backupJson}`);
+      // Also back up our yarn.lock
+      await fse.renameSync(`${root}/yarn.lock`, `${root}/${backupYarnLock}`);
+    } catch (error) {
+      console.error(error);
+    }
+  });
+
+  beforeEach(async () => {
+    // Create a copy package.json so we can use a fresh copy for each test
+    await fse.copyFile(`${root}/${backupJson}`, `${root}/package.json`);
+    // Create a copy of the yarn.lock
+    await fse.copyFile(`${root}/${backupYarnLock}`, `${root}/yarn.lock`);
+  });
+
+  afterEach(async () => {
+    // Remove package.json and yarn.lock after each test completes
+    await fse.remove(`${root}/package.json`);
+    await fse.remove(`${root}/yarn.lock`);
+  });
+
+  afterAll(async () => {
+    try {
+      // Restore our package.json from backup
+      await fse.renameSync(`${root}/${backupJson}`, `${root}/package.json`);
+      // Restore our yarn lock
+      await fse.renameSync(`${root}/${backupYarnLock}`, `${root}/yarn.lock`);
+    } catch (err) {
+      console.error(err);
+    }
+  });
+
+  it("expects to find a backup.json and package.json", async () => {
```

Comment:
Should we be checking for `yarn.lock` to exist here as well?

Reply:
Jeffery, your eyes are impeccable - great catch - I'll add that in now